Havit.Blazor.Components.Web.Bootstrap/Forms/Autosuggests/Internal/HxAutosuggestInternal.razor

```diff
@@ -31,27 +31,27 @@
         }
         @if (EnabledEffective)
         {
-            <div class="hx-autosuggest-input-icon">
-                @if (_dataProviderInProgress)
+
+            @if (_dataProviderInProgress)
+            {
+                <div class="hx-autosuggest-input-icon">
+                    <span class="spinner-border spinner-border-sm" role="status" aria-hidden="true"></span>
+                </div>
+            }
+            else if (!EqualityComparer<TValue>.Default.Equals(Value, default))
+            {
+                if (ClearIconEffective is not null)
                 {
-                    <div>
-                        <span class="spinner-border spinner-border-sm" role="status" aria-hidden="true"></span>
+                    <div role="button" @onclick="HandleCrossClick" @onclick:stopPropagation class="hx-autosuggest-input-icon">
+                        <HxIcon CssClass="hx-autosuggest-input-icon-clear" Icon="ClearIconEffective" />
                     </div>
                 }
-                else if (!EqualityComparer<TValue>.Default.Equals(Value, default))
-                {
-                    if (ClearIconEffective is not null)
-                    {
-                        <div role="button" @onclick="HandleCrossClick" @onclick:stopPropagation>
-                            <HxIcon CssClass="hx-autosuggest-input-icon-clear" Icon="ClearIconEffective" />
-                        </div>
-                    }
-                }
-                else if (SearchIconEffective is not null)
-                {
-                    <HxIcon CssClass="hx-autosuggest-input-icon-search" Icon="@SearchIconEffective" />
-                }
-            </div>
+            }
+            else if (SearchIconEffective is not null)
+            {
+                <HxIcon CssClass="hx-autosuggest-input-icon hx-autosuggest-input-icon-search" Icon="@SearchIconEffective" />
+            }
```

Comment:
However, this change removes the wrapping `div.hx-autosuggest-input-icon` that was around **any** icon, not just _Clear_. This alters the structure, so that some component icons have a `div` around them, while others don’t (and have `hx-autosuggest-input-icon` directly on them). This doesn't seem consistent to me.

If we need to add a handler to the wrapping `div`, I would either create a single `div` and one `HxIcon` with everything controlled by conditions, or I would create several branches with `div` + `HxIcon`, ensuring that each branch has a consistent structure of `div` + `HxIcon`.

@crdo?

Reply:
@hakenr @petrhank the wrapper is there because of the positioning and should be around all of the icons. I have no preferece between single div or wrapp all, let`s just be consistent and do it same way in SearchBox and Autosuggest.
I also prefere naming the handler ClearInputAsync over HandleCrossClick.